README.md

```diff
@@ -1,16 +1,130 @@
+> particular author. Check out the documentation to see all the ways to customize
+> your commands.
+
+## Focus on Certain Changes with `git diff`
+
+When examining a repository's history, it's not always apparent _what_ or _how_
+something changed. Fortunately, Git gives us a command to look at changes right
+next to each other:
+
+```bash
+$ git diff
+```
+
+This command will show us the differences between previous commits and current
+work, or between different branches, or anything else we need to track changes
+between. Similar to `git log`, we can customize the command with the parameters
+of what we're comparing.
+
+`git diff` gives us another layer of granularity when it comes to a Git
+repository's history. We can look at the _log_ to see where changes were
+committed and then look at the _diff_ to see exactly what those changes were.
+
+## Unstage Changes with `git reset`
+
+It happens to all developers at some point: we make quick changes, add and
+commit them to the repository. But before we push our changes, we realize we
+committed something incomplete or made an error in our commit message. How do we
+go back and fix it? We use the `reset` command to bring our work back the state
+before the commit:
+
+```bash
+$ git reset
+```
+
+This command also comes with various flags we can add to specify which point we
+want to go back to. For example, if we just want to undo the last commit, we use
+the command:
+
+```bash
+$ git reset --soft HEAD~1
+```
+
+Now Git rewinds to the way our code was before the last commit. If we need to go
+even further back, we could ask Git to take us, for example, three commits back
+in the repo history:
+
+```bash
+$ git reset --soft HEAD~3
+```
+
+> **Tip** The `--soft` tag keeps the changes we've made since last commit in our
+> working state so that we can fix, add to or remove them manually. If we know we
+> want to clear those changes entirely, we can use the `--hard` flag instead.
+> Check the documentation to see all the ways to customize your command.
+
+## Revert Changes with `git checkout`
+
+We discussed the command `git checkout` earlier when describing how to switch
+between different branches. We can also use it as a way to clear unwanted
+changes from our working branch. After all, one of the best things about using
+Git to organize and keep track of our work is that it lets us explore and
+experiment without running the risk of accidentally ruining our work. Maybe
+you've already encountered a situation where you started to code a new feature
+or solution but after a while realized you didn't want to continue it. Instead,
+you wanted to clear it all and start back at the beginning. Run:
+
+```bash
+$ git checkout .
+```
+
+... and you'll have a nice, clean working branch.
+
+If we want to limit our cleaning to a single file, we can specify that file
+instead of cleaning out everything:
+
+```bash
+$ git checkout index.html
+```
+
```

Comment:
Probably worth mentioning `git checkout -p` since we talked about it for `git commit`